### PR TITLE
v2.20.0

### PR DESCRIPTION
**Pull request by CiprianDraghici**
v2.20.0

### CODE REVIEW

File: src/apiCalls/transactions/sendSignedBatchTransactions.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  batchId?: string | null;
+  data?: BatchTransactionsResponseType;
+};
+
+export async function sendSignedBatchTransactions({
+  transactions,
+  sessionId
+}: SendBatchTransactionsPropsType) {
+  const address = addressSelector(store.getState());
+  const { apiAddress, apiTimeout } = networkSelector(store.getState());
```
Comment (arhtudormorar) [inline src/apiCalls/transactions/sendSignedBatchTransactions.ts:31]: if (!address) {
return;
}

File: src/utils/transactions/batch/generateBatchTransactionsGrouping.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import { generateBatchTransactionsGrouping } from './generateBatchTransactionsGrouping'; // Replace 'your-module' with the actual module path
+
+describe('generateBatchTransactionsGrouping', () => {
```
Comment (arhtudormorar) [inline src/utils/transactions/batch/generateBatchTransactionsGrouping.spec.ts:3]: file name .test.ts

File: src/services/transactions/sendBatchTransactions.spec.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
+import { sendBatchTransactions } from './sendBatchTransactions';
+import { addressSelector } from 'reduxStore/selectors';
+import { store } from 'reduxStore/store';
+import { getWindowLocation } from 'utils/window/getWindowLocation';
+import { signTransactions } from './signTransactions';
+import { transformTransactionsToSign } from './utils/transformTransactionsToSign';
+
+jest.mock('reduxStore/selectors', () => ({
```
Comment (arhtudormorar) [inline src/services/transactions/sendBatchTransactions.spec.ts:8]: file name .test.ts

File: src/apiCalls/transactions/sendSignedBatchTransactions.ts (hunk 1 above)
Comment (CiprianDraghici) [reply src/apiCalls/transactions/sendSignedBatchTransactions.ts:31]: Not necessary. This function will be called only in a logged-in state

**Review by razvantomegea — APPROVED** [review]

**Review by ioandanpopa — APPROVED** [review]

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**